compiler/renderer.nim

```diff
@@ -839,12 +841,12 @@ proc gident(g: var TSrcGen, n: PNode) =
       t = tkSymbol
   else:
     t = tkOpr
-  put(g, t, s, if n.kind == nkSym and renderSyms in g.flags: n.sym else: nil)
+  var suffix = ""
   if n.kind == nkSym and (renderIds in g.flags or sfGenSym in n.sym.flags or n.sym.kind == skTemp):
+    suffix = genSymSuffix(n.sym)
     when defined(debugMagics):
-      put(g, tkIntLit, $n.sym.id & $n.sym.magic)
-    else:
-      put(g, tkIntLit, $n.sym.id)
+      suffix.add "_" & $n.sym.magic
+  put(g, t, s & suffix, if n.kind == nkSym and renderSyms in g.flags: n.sym else: nil)
```

Comment:
The old way didn't do a string concat for every rendered identifier.

Reply:
done